### ISSUE TITLE
Failed to create cluster -  Valid IP address error

### ISSUE THREAD

**Issue by deven7aq** [kubernetes-sigs/kind#1657]
$ kind create cluster  --name Cluster2 --kubeconfig cluster1.yaml -v 1
Creating cluster "Cluster2" ...
DEBUG: docker/images.go:58] Image: kindest/node:v1.18.2@sha256:7b27a6d0f2517ff88ba444025beae41491b016bc6af573ba467b70c5e8e0d85f present locally
 ✓ Ensuring node image (kindest/node:v1.18.2) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✗ Starting control-plane 🕹️ 
ERROR: failed to create cluster: failed to init node with kubeadm: command "docker exec --privileged Cluster2-control-plane kubeadm init --ignore-preflight-errors=all --config=/kind/kubeadm.conf --skip-token-print --v=6" failed with error: exit status 1
Command Output: I0606 06:58:49.949593     164 initconfiguration.go:200] loading configuration from "/kind/kubeadm.conf"
[config] WARNING: Ignored YAML document with GroupVersionKind kubeadm.k8s.io/v1beta2, Kind=JoinConfiguration
I0606 06:58:49.952645     164 interface.go:400] Looking for default routes with IPv4 addresses
I0606 06:58:49.952655     164 interface.go:405] Default route transits interface "eth0"
I0606 06:58:49.952714     164 interface.go:208] Interface eth0 is up
I0606 06:58:49.952753     164 interface.go:256] Interface "eth0" has 3 addresses :[172.18.0.2/16 fc00:f853:ccd:e793::2/64 fe80::42:acff:fe12:2/64].
I0606 06:58:49.952773     164 interface.go:223] Checking addr  172.18.0.2/16.
I0606 06:58:49.952782     164 interface.go:230] IP found 172.18.0.2
I0606 06:58:49.952794     164 interface.go:262] Found valid IPv4 address 172.18.0.2 for interface "eth0".
I0606 06:58:49.952802     164 interface.go:411] Found active IP 172.18.0.2 
hostport Cluster2-control-plane:6443: host 'Cluster2-control-plane' must be a valid IP address or a valid RFC-1123 DNS subdomain
k8s.io/kubernetes/cmd/kubeadm/app/util.ParseHostPort
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/endpoint.go:112
k8s.io/kubernetes/cmd/kubeadm/app/util/config.SetClusterDynamicDefaults
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubeadm/app/util/config/initconfiguration.go:157

- kind version: v0.8.1
- Kubernetes version: (use `kubectl version`):
- Docker version: 19.03.6:
- OS : Ubuntu 18.04

**Comment by BenTheElder** [comment]
We've already "fixed" this in head.

TLDR: you can't use capital names.

Longer: Kubernetes upstream has a bug which is now not fixable due to compatibly reasons apparently that validating DNS subdomains against the RFC only allows lowercase. kind only intented to allow cluster names that are a subset of what is valid upstream, so we now dissallow capitals too. This shouldn't have worked before.

**Comment by BenTheElder** [comment]
Sorry about that, you'll need to use a lowercase
/triage duplicate
https://github.com/kubernetes-sigs/kind/issues/1601
